### DISCUSSION TITLE
Neurodocker roadmap

### DISCUSSION THREAD

**Discussion by djarecka** [ReproNim/neurodocker#522] [Announcements]
recently @satra asked @sooyounga and I to help with maintenance. We have been mostly cleaning some old issues, but we can try to handle some code updates.
However we are thinking that it would be nice to meet and chat with long-term developer and users about the project future. 

@kaczmarj , @stebo85 , @Remi-Gau - would you be interested in the meeting? If yes, could you please comment which weeks/months in this spring would possibly work (or do not work), so I could create a poll to find a time that works for everyone.

**Comment by kaczmarj** [comment]
yes i'd like to attend. i can make any date/time this spring work.

**Comment by stebo85** [comment]
yes, would love to attend :) A time that is Australia-time AEST friendly would be great.

**Comment by Remi-Gau** [comment]
I would like to attend. My agenda is fairly flexible and I should be either in Brussels or Montreal in the coming months.

**Reply by djarecka** [reply]
@Remi-Gau - when are you planning to be in Montreal? It would be easier to arrange a meeting in reasonable time of the day for everyone when you are in Montreal.

**Reply by Remi-Gau** [reply]
I don't know yet. I should have an answer any day now. Will keep you posted.

**Reply by Remi-Gau** [reply]
OK.
I am now in Canada for about a month, so we could bring this back to the front burner.

**Reply by stebo85** [reply]
From 7pm onwards Canada time would be perfect for me - I can make that work most days of the week :)

**Comment by djarecka** [comment]
I'm sorry that I didn't follow with the discussion, but let's try to meet in August. I've just met Remi in Montreal, and seems like he is here till the end of the summer. So I will create a doodle poll for days in August for 6pm and 7pm ET (8am and 9am AEST)

**Reply by kaczmarj** [reply]
That's great. I will be in Copenhagen August 24 to September 3, so I would prefer if we met before August 24.

**Comment by djarecka** [comment]
please fill the poll if you would like to join: https://doodle.com/meeting/participate/id/ej8BNKyb
If we can't find anything with these days, I will create a new link.

Adding more people from repronim that might be interested: @dnkennedy @yarikoptic @satra 
Feel free to add anyone who might be interested

**Reply by stebo85** [reply]
thank you so much for choosing times where I can join =)

**Comment by djarecka** [comment]
@satra - once you have a chance, could you fill the poll:  https://doodle.com/meeting/participate/id/ej8BNKyb

**Comment by djarecka** [comment]
just a reminder that the meeting will be on Monday 6pm ET time. 

@Remi-Gau - I've just realized that you never answered... Can you make this time? or should we try to find new time in Sept?

**Reply by Remi-Gau** [reply]
yup
can do
can you send an invite on: remi dot gau at gmail dot com

**Comment by djarecka** [comment]
A link for the notes from the discussion: https://docs.google.com/document/d/1x5rgzFgqFJ6YYPzMNOjEnMca3b6xjUci4ndR-cdgnl4/edit

**Comment by djarecka** [comment]
I want to continue here the discussion on what should be the main mission for the project. As a starting point I will just copy notes from the document, but would be great to continue this discussion:

- How much we should create more recipes in neurodocker vs upstreaming to conda, easybuild, neurodebian etc. (more maintainers)
      - Focus on installation of binary builds that are easier to maintain
      - Seek to upstream building from source upstream/distributions
      - Consider additional distributions (e.g. [easybuilds](https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs); for the discussion specifically on easubuilds and EESSI project go to #546 ) 
- Providing reproducibility (e.g.  problem with ants)
      - Establish an archive of (some) relevant binary builds. Problem: some might be requested to be removed

**Comment by kaczmarj** [comment]
i agree we should be focusing on the installation of binary builds. binary installations have fewer edge cases than source builds and in general are much easier to manage / maintain. 

regarding uploading to conda or other distributions, i am not sure. i'd love to hear input from others on this topic.

the idea of using easybuild for source builds is fantastic, and recipes to build AFNI, ANTs, FSL, FreeSurfer, SPM (and maybe others) are already included. we can add to those recipes. neurodocker would then be a frontend to easybuild, abstracting away its complexities. my main concern is the amount of effort it would take to accomplish this. though, easybuild would probably solve one of our biggest and most longstanding problems -- building complex containers with multiple softwares with dependencies that might clash.

providing an archive of relevant binary builds could be difficult to maintain over the long term. i don't know how long this effort would last, because one would have to be devoted to uploading binaries somewhere. my worry is that this effort would fizzle out after some time.
